### DISCUSSION TITLE
The version 3 does not seem to offer the same settings as v2

### DISCUSSION THREAD

**Discussion by ale-bast** [johnfactotum/foliate#1175] [Q&A] [answered]
I primarily used foliate v2 to read e-books in a foreign language and used it to translate certain sentences or read a specific word with gtts-cli using the script in the wiki. It used to be possible to specify in the settings to use the script when one word was selected, and to automatically translate when multiple words where selected while remembering the language used to translate last time so as to not have to select it every time. Also opening the last book being read was nice.

However now it seems those settings are gone, while all the other problems I had with foliate seem to have been fixed, especially the app crashing when going from a book to the library and opening another book in more than 10 seconds(if done fast enough it did not crash).

Given how much better everything else is now I would just like to know if there is still a way to have those settings back so i don't go back to v2?
Thank you for all the work you put in foliate it has been extremely useful

**Comment by johnfactotum** [comment, marked as answer]
No, it's not possible. Currently those features are missing in v3.

> I primarily used foliate v2 to read e-books in a foreign language and used it to translate certain sentences or read a specific word with gtts-cli [...]

As a workaround, you can try binding a global shortcut on your desktop to launch custom tools with the contents of the primary selection using `wl-paste` or `xclip`.

The downside of this is that you need to press a shortcut after selecting (but note that `wl-paste`, for example, has a `--watch` option). The upside is that it will also work in other apps, not just Foliate.

> Also opening the last book being read was nice.

You can kind of work around this with a custom script. First find the last modified JSON file in Foliate's data directory. The file name is URI component enocded. Decode the filename and you will get the last book's identifier.

From here either read `library/uri-store.json` to get the URI stored by Foliate or query the file indexer, e.g. Tracker, to get the file's location. You can then open the file.

Strictly speaking this is the book where you last navigated or annotated, which is usually but not necessarily the last opened or closed file.

**Reply by ale-bast** [reply]
Ok, thank you
